### DISCUSSION TITLE
Foundry reinstallation everytime i start a new terminal

### DISCUSSION THREAD

**Discussion by p-z-val** [Cyfrin/foundry-full-course-cu#585] [General]
I had installed foundry and it was working fine. Idk but probably after some change, now whenever i start a new terminal or session, i have to reinstall foundry in that terminal everytime. Why is that and how to fix it?

**Comment by theirrationalone** [comment]
are you using wsl??

**Reply by theirrationalone** [reply]
If You're using wsl then don't forget to open remote window by hitting that bottom left wsl icon. ✅ (i might be doing it right)

try to update wsl:

```terminal
sudo apt update
```

or you might forgetting to do this... just after curl command...

```terminal
source /Users/<username>/.bashrc
```

after executing above command you should then do that `foundryup` i.e.,

```terminal
foundryup
```

then after this, You will have all foundry binaries(tools) i.e., forge, cast, chisel, in all sessions/terminals.

**Reply by p-z-val** [reply]
I am using MAC terminal. And i am writing the above steps everytime i open a new bash shell. Doesn't change anything. I have to write those 3 lines of code in bash shell everytime i open a new shell.

I had made a couple of changes to vs code, but i am not sure which change is causing this. So far, I think changing the default shell to bash from zsh might have caused this

**Comment by itznishant** [comment]
Do refer this section in the main codebase for the installation video steps based on your  environment. 

https://github.com/Cyfrin/foundry-full-course-f23#lesson-6-foundry-simple-storage

Try to check if this helps out with the issue.

**Reply by p-z-val** [reply]
okay

**Reply by p4y4b13** [reply]
do you fixed this error dude? I am also facing the same issue

**Comment by theirrationalone** [comment]
You need to update bash Path... 
Note: <you_username> means patrick, mariah, alice, bob, charlie, duke, etc. 
Ex: `/Users/<your_username>/.bashrc` === `/Users/alice/.bashrc`

Patrick's words... From lesson 7(a part of that) 

You can mitigate this issue by executing this command:
Two methods we have.... 
First:  Open terminal (navigated to your project's directory). 

```terminal
source /Users/<your_username>/.bashrc
```

If first command didn't worked then here is second method... 

Second: Open terminal (navigated to your project's directory). Then, 

1. Type: `cd` (in your terminal to go to your home directory)  Hit Enter. 
2. Type: `echo "source /Users/<your_username>/.bashrc" >> . bash_profile` Hit Enter. 

Done.

**Reply by p4y4b13** [reply]
Thanks dude!, It is working now

**Reply by NeoRusi** [reply]
I have done the same with zsh but It worked until I didn't kill the terminal 
first time the foundry was installed globaly just by running  the terminal not in a certain directory 
```curl -L https://foundry.paradigm.xyz | bash```
then I ran the command like this 
```source /var/folders/s_/3hsn_sqj7599s946j60ph22m0000gn/T/vscode-zsh/.zshenv```
then I run
 ```cd```
then 
```echo "source /var/folders/s_/3hsn_sqj7599s946j60ph22m0000gn/T/vscode-zsh/.zshenv" >> . zsh_profile```
and it outputs this 
```zsh: is a directory: .```

**Reply by theirrationalone** [reply]
First of all, We're executing & setting up the foundry with bash

It's showing that `zsh` is a directory not a `rc config file`, Please reach to foundry setup directory and search for `.bashrc`...

See i'm on windows and i'm on foundry setup directory...

![bashrc config](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/4c20983f-d642-45c5-b6b1-7641d8f8353b)

Please Find directory like this on your macos.... and eventually you will get the idea what to do then after...

**Reply by NeoRusi** [reply]
Sorry but I don't actually understand how to work with files 
```
Detected your preferred shell is zsh and added foundryup to PATH. Run 'source /var/folders/s_/3hsn_sqj7599s946j60ph22m0000gn/T/vscode-zsh/.zshenv' or start a new terminal session to use foundryup.
Then, simply run 'foundryup' to install Foundry.
```
and I don't understand how to switch preferred shell from zsh to bash

**Reply by NeoRusi** [reply]
Also I am on Mac and there is no such file manager

**Reply by theirrationalone** [reply]
Sorry for the late reply buddy @NeoRusi !, 

Please Follow this Image: If it can help you anyway...

![bash for macos](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/35253739-7421-4c70-bfb8-2a07db13a4af)

Actually you need to have your bash terminal then you'll be good to go....

If Above Steps didn't helped you then Please follow this [resource](https://www.cyberciti.biz/faq/change-default-shell-to-bash-on-macos-catalina/), Since you're using `macOS`.

Please don't feel discouraged, Even if above triages don't help you, In the next comment you can tag Patrick also.

**Reply by NeoRusi** [reply]
Bro thank you very much and good luck with the raffle section of the foundry dev course

**Reply by 0xTagger** [reply]
<img width="967" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/a25a9ba2-b240-4d11-ae98-0ff76a44bfc3">

hey @theirrationalone i've been having the same issue where when I open a new terminal in VSCode I have to reinstall Foundry every single time. It was working for nearly 2 months no issue and happened out of no where. I have tried all I can find to resolve this issue including your suggestions. I don't have a bash_profile file, is this the issue?  any feedback from you is much appreciated. Thanks!!

EDIT: I figured it out :) it was because there wasn't a .bash_profile file on my mac and the Foundry path wasn't in a .bash_profile file.
